src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2603,9 +2617,21 @@ BuildRoutesForInsert(Query *query, DeferredErrorMessage **planningError)
 		if (partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
 			DISTRIBUTE_BY_RANGE)
 		{
+			cacheEntry = GetCitusTableCacheEntry(distributedTableId);
```

Comment:
`cacheEntry` shouldn't be assigned conditionally, since we unconditionally call `FindShardInterval`

Reply:
It looks like we already get it at the top of `BuildRoutesForInsert`. Do you think should we get it here as well, or removing  is fine?

Reply:
Removing is fine